src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,289 @@
+        System.out.println(SECTION_DIVIDER);
+    }
+
+    public static Todo addTodoToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("todo") + INDEX_TODO;
+        Todo todoAdded = new Todo(userInput.substring(descriptionIndex));
+        TaskArray.add(todoAdded);
+        return todoAdded;
+    }
+
+    public static Event addEventToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("event") + INDEX_EVENT;
+        int byIndex = userInput.indexOf("/at");
+        Event eventAdded = new Event(userInput.substring(descriptionIndex, byIndex-1), userInput.substring(byIndex + 4));
+        TaskArray.add(eventAdded);
+        return eventAdded;
+    }
+
+    public static Deadline addDeadlineToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("deadline") + INDEX_DEADLINE;
+        int byIndex = userInput.indexOf("/by");
+        Deadline deadlineAdded = new Deadline(userInput.substring(descriptionIndex, byIndex-1), userInput.substring(byIndex + 4));
+        TaskArray.add(deadlineAdded);
+        return deadlineAdded;
+    }
+
+    public static void markTaskAsDone(String userInput) {
+        userInput = userInput.replaceAll("[^0-9]", "");
+        try {
+            int taskNo = Integer.parseInt(userInput);
+            if (taskNo > TaskArray.size()) {
+                System.out.println("You only have " + TaskArray.size() + " task(s)!");
+            } else {
+                TaskArray.get(taskNo - 1).markAsDone();
+                System.out.println("Nice! I have marked this task as done:" + TaskArray.get(taskNo - 1));
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid Input! Input format should have an integer e.g. done 2");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid Input! Integer cannot be 0!");
+        }
+        System.out.println(SECTION_DIVIDER);
+    }
+
+    private static void deleteTask(String userInput) {
+        userInput = userInput.replaceAll("[^0-9]", "");
+        try {
+            int taskNo = Integer.parseInt(userInput);
+            if (taskNo > TaskArray.size()) {
+                System.out.println("You only have " + TaskArray.size() + " task(s)!");
+            } else {
+                System.out.println("Noted. I've removed this task: " + System.lineSeparator() + TaskArray.get(taskNo - 1));
+                TaskArray.remove(taskNo - 1);
+                System.out.println("Now you have " + TaskArray.size() + " tasks in the list.");
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid Input! Input format should have an integer e.g. done 2");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid Input! Integer cannot be 0!");
+        } finally {
+            System.out.println(SECTION_DIVIDER);
```

Comment:
Good job in attempting to create a [clear path of execution](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent), with the possible exceptions.